doctor/router_v3.py

```diff
@@ -0,0 +1,128 @@
+import functools
```

Comment:
Will likely rename this file to something else before v3.0.0 since we don't have a router class anymore.